src/Tribe/Rewrite.php

```diff
@@ -407,19 +407,25 @@ public function get_canonical_url( $url, $force = false ) {
 		$query         = (string) parse_url( $url, PHP_URL_QUERY );
 		wp_parse_str( $query, $query_vars );
 
-		// Remove the `paged` query var if it's 1.
 		if ( isset( $query_vars['paged'] ) && 1 === (int) $query_vars['paged'] ) {
+			// Remove the `paged` query var if it's 1.
 			unset( $query_vars['paged'] );
 		}
 
 		ksort( $query_vars );
 
 		$our_rules          = $this->get_handled_rewrite_rules();
 		$handled_query_vars = $this->get_rules_query_vars( $our_rules );
+		$handled_post_types = $this->get_post_types();
 
 		if (
+			// The rules we handle should not be empty.
 			empty( $our_rules )
-			|| ! in_array( Arr::get( $query_vars, 'post_type', 'post' ), $this->get_post_types(), true )
+			|| ! (
+				// Supported post types should be either keys or values, of the `post_type` argument, in the query vars.
+				count( array_intersect_key( array_flip( $handled_post_types ), $query_vars ) )
+				|| in_array( Arr::get( $query_vars, 'post_type', 'post' ), $handled_post_types, true )
+			)
```

Comment:
:no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: d82c7797a54ed86a0c7550f164149395</sub>